src/java.desktop/share/classes/javax/swing/plaf/basic/BasicComboBoxUI.java

```diff
@@ -1118,12 +1118,7 @@ else if (comboBox.getModel().getSize() > 0) {
     // This is currently hacky...
     @Override
     public int getAccessibleChildrenCount(JComponent c) {
-        if ( comboBox.isEditable() ) {
-            return 2;
-        }
-        else {
-            return 1;
-        }
+        return 2;
```

Comment:
I think the logic here has to be equal to the logic in `getAccessibleChild `, otherwise `getAccessibleChildrenCount` returns 2 even when `getAccessibleChild(..., 1)` returns null.

Reply:
Can you please elaborate more or what logic can be added here?

Reply:
I mean `getAccessibleChild(..., 1)` can still return null (when both `if`s of `case 1` resolve to false). Let's consider the following code:
```java
public void doSmthWith(JComponent component, BasicComboBoxUI ui) {
    if (ui.getAccessibleChildrenCount(component) > 1) {
        // well, we have at least 2 components, so we can expect to get a valid object below
        final var ac = ui.getAccessibleChild(component, 1)
                                  .getAccessibleContext(); // <= possible NPE
        // ...
    }
}
```

I think we have to return 1 from `getAccessibleChildrenCount(...)` for the cases when `getAccessibleChild(..., 1)` returns `null`

Reply:
I think it is not needed to return 1 from `getAccessibleChildrenCount(...)`.  `getAccessibleChildrenCount(...)` is used to get the exact number of accessible child for a component and afterwards `getAccessibleChild(..., index)` method can be invoked to get the child.

It should be ok to return null if the switch case condition fails or index doesn't match with case statement. Although chances of returning null is very unlikely for index 0 and 1.

Reply:
please double-check how it will work in Windows, it might be possible that jaws expects 1.

Reply:
> please double-check how it will work in Windows, it might be possible that jaws expects 1.

@mrserb I observed the difference in announcement with JAWS with the fix. 
Since the magnifier is used only for macos, proposed fix is now handled in AquaComboBoxUI instead of BasicComboBoxUI. Changes in BasicComboBoxUI is reverted back.